### DISCUSSION TITLE
Suddenly hitting rate-limit 429

### DISCUSSION THREAD

**Discussion by Glutch** [vercel/community#1182] [Help] [answered]
Hello!

After a viral reddit post my site is hitting rate limit 429 and i am missing a lot of sales. 
This is very strange since i've had many viral posts before and this has never been an issue. (I do have pro plan)
This current post is also significantly less viral than the others, currently 20 users the last 5 minutes according to google analytics

I've checked the logs and removed at least 6 different api calls, even completely removed the ability to sign in, since that is not mandatory to purchase. I am still getting rate limited and i have no idea what to do. I have contacted vercel sales team and help, but they say 24-48 hours, at that point my viral thead has already died down

Do you have any suggestions? Or do you know a shortcut to reach support?

Any help appreciated

**Comment by leerob** [comment, marked as answer]
Hey there, give this a read: https://vercel.com/guides/what-should-i-do-if-i-receive-a-429-error-on-vercel

And let me know if you have any questions.

**Reply by Glutch** [reply]
I've read that and purged all my projects to use the absolute minimum amount of api calls, still getting 429

**Reply by Glutch** [reply]
I just started tracking logs, and i seem to be getting a request every couple of ms, to my images on the site.
on mysite.com/catalog i show 100 images, hosted by aws s3 & next/image
Could this be the issue?

<img width="1159" alt="image" src="https://user-images.githubusercontent.com/11299527/207415118-c8b94f0e-25e0-4aef-afc8-5bd906128406.png">

**Reply by Glutch** [reply]
18000 logs in 15~ minutes

<img width="1190" alt="image" src="https://user-images.githubusercontent.com/11299527/207416211-bc5d2194-4089-46e3-a019-ded1e360afc5.png">

**Reply by Glutch** [reply]
i think that was the issue... when i changed from image/nextjs to normal img, it's real quiet!

<img width="1189" alt="image" src="https://user-images.githubusercontent.com/11299527/207418745-1e1ab5be-93b8-4091-a458-89228583fb01.png">

**Reply by amyegan** [reply]
I'm glad you got it working! Are you on Next.js 12, or have you moved to version 13?

**Reply by Glutch** [reply]
> I'm glad you got it working! Are you on Next.js 12, or have you moved to version 13?

I hope so! However the 429 errors has not stopped, hopefully soon (?)
<img width="1461" alt="image" src="https://user-images.githubusercontent.com/11299527/207420826-87933ac5-62e2-4db3-9bbf-4ad21b3e32c5.png">

**Reply by Glutch** [reply]
For future humans, the limit released 20 minutes after solving the issue

(filter: FUNCTION_RATE_LIMIT)
<img width="1187" alt="image" src="https://user-images.githubusercontent.com/11299527/207421686-cecedb8c-7b4a-4e6e-8660-cd0d4a6854e6.png">
(filter: All logs)
<img width="1187" alt="image" src="https://user-images.githubusercontent.com/11299527/207421731-d7ef9a63-e75a-4bd8-b0bf-8fa6351aefef.png">
